driver-core/src/main/java/com/datastax/driver/core/CodecRegistry.java

```diff
@@ -0,0 +1,760 @@
+        }
+        return this;
+    }
+
+    /**
+     * Register the given codec with this registry as an "overriding" codec.
+     * <p>
+     * This codec will only be used if the value being serialized or deserialized
+     * belongs to the specified qualified name.
+     * <p>
+     * Qualified names shoud be specified in the form of {@code keyspace.table.column} or {@code keyspace.usertype.field}.
+     * <p>
+     * <strong>IMPORTANT</strong>: Overriding codecs can only be used in conjunction with {@link GettableData} instances,
+     * such as{@link BoundStatement}s and {@link Row}s.
+     * It <em>cannot</em> be used with other instances of {@link Statement}, such
+     * as {@link SimpleStatement} or with the {@link com.datastax.driver.core.querybuilder.QueryBuilder query builder}.
+     * <p>
+     * <em>For this reason, if you plan to use overriding codecs, be sure to only use {@link PreparedStatement}s.</em>
+     *
+     * @param codec The overriding codec.
+     * @param fqdn The fully-qualified name of the table column or UDT field to which the overriding codec applies;
+     * if {@code null}, no overriding codec will be located and the globally-registered codec will be returned instead.
+     * @return this Builder (for method chaining).
+     */
+    public CodecRegistry register(TypeCodec<?> codec, String fqdn) {
+        checkNotNull(fqdn, "Parameter fqdn cannot be null");
+        overrides.put(fqdn, codec);
+        return this;
+    }
+
+    /**
+     * Returns a {@link TypeCodec codec} that accepts the given value.
+     * <p>
+     * This method takes an actual Java object and tries to locate a suitable codec for it.
+     * For this reason, codecs must perform a {@link TypeCodec#accepts(Object) "manual" inspection}
+     * of the object to determine if they can accept it or not, which, depending on the implementations,
+     * can be an expensive operation; besides, the resulting codec cannot be cached.
+     * Therefore there might be a performance penalty when using this method.
+     * <p>
+     * Furthermore, this method would return the first matching codec, regardless of its accepted {@link DataType CQL type}.
+     * For this reason, this method might not return the most accurate codec and should be
+     * reserved for situations where the runtime object to be serialized is not available or unknown.
+     * In the Java driver, this happens mainly when serializing a value in a {@link SimpleStatement}
+     * or in the {@link com.datastax.driver.core.querybuilder.QueryBuilder}, where no CQL type information
+     * is available.
+     * <p>
+     * Regular users should avoid using this method.
+     * <p>
+     * Codecs returned by this method are NOT cached.
+     *
+     * @param value The value the codec should accept; must not be {@code null}.
+     * @return A suitable codec.
+     * @throws CodecNotFoundException if a suitable codec cannot be found.
+     */
+    public <T> TypeCodec<T> codecFor(T value) {
```

Comment:
cool, that makes perfect sense and I do agree that the symmetry with the get methods is nice.